GHCN.py

```diff
@@ -0,0 +1,1044 @@
+  atts        = None # dictionary with PyGeoData variable attributes
+  prefix      = '' # file prefix for source file name (used with station ID)
+  fileext     = '.dly' # file name extension (used for source files)
+  # type specific
+  datatype    = '' # defined in child class; type of source file
+  recordunits = '' # units used for data  (in source files)
+  units       = '' # units after conversion
+  scalefactor = 1 # constant scaling factor for conversion 
+  offset      = 0 # constant offset for conversion
+  dtype       = 'float32' # data type used for data in source files
+  encoding    = 'UTF-8' # file encoding (used in source files)
+  missing     = '' # string indicating missing value
+  mflag        = '' # legal data flags (case sensitive)
+  qflag        = ''
+  sflag        = ''
+  varmin      = 0. # smallest allowed value in data
+  varmax      = 0. # largest allowed value in data
+  # inferred variables
+  variable    = '' # alias for name
+  filepath    = '' # inferred from prefix
+  
+  def __init__(self, **kwargs):
+    super(VarDef,self).__init__(**kwargs)
+    self.variable = self.name
+    self.filepath = '{0:s}{1:s}'.format('{:s}',self.fileext)
+    
+  def convert(self, data): 
+    if self.scalefactor != 1: data *= self.scalefactor;
+    if self.offset != 0: data += self.offset
+    return data
+  
+  def getKWargs(self, *args):
+    ''' Return a dictionary with the specified arguments and their values '''
+    if len(args) == 0: 
+      args = ['variable', 'units', 'varmin', 'varmax', 'missing', 'mflag', 'qflag', 'sflag', 'dtype', 'encoding']
+    kwargs = dict()
+    for arg in args:
+      kwargs[arg] = getattr(self,arg)
+    return kwargs
+  
+# definition for precipitation files
+class PrecipDef(VarDef):
+#   recordunits = 'mm' # actually 'mm/day' but reported as 'mm'
+#   units       = 'kg/m^2/s' # units after scaling (SI)
+  scalefactor = .1/(24.*60.*60.) # convert from mm/day to mm/s 
+  offset      = 0 
+  datatype    = 'precip'
+  title       = 'GHCN Precipitation Records'
+  missing     = '-9999' # string indicating missing value (apparently not all have 'M'...)
+  MFLAG1      = ' BDHKLOPTW' # legal data flags (case sensitive; 'M' for missing should be screened earlier)
+  QFLAG1      = ' DGIKLMNORSTWXZ'
+  SFLAG1      = ' 067AaBbCEFGHIKMNQRrSsTUuWXZz'
+  varmin      = 0. # smallest allowed value in data
+  varmax      = 1.e4 # largest allowed value in data
+  
+# definition for temperature files
+class TempDef(VarDef):
+#   recordunits = u'°C'
+#   units       = 'K'
+  scalefactor = 0.1 
+  offset      = 273.15 # convert to Kelvin 
+  datatype    = 'temp'
+  title       = 'GHCN Temperature Records'
+  missing     = '-9999' # string indicating missing value
+  MFLAG1      = ' BDHKLOPTW' # legal data flags (case sensitive; 'M' for missing should be screened earlier)
+  QFLAG1      = ' DGIKLMNORSTWXZ'
+  SFLAG1      = ' 067AaBbCEFGHIKMNQRrSsTUuWXZz'

```

Comment:
If the flags are the same for precip and temp, you can add them to the parent class.